patches/server/0986-FOV-Scale-Implementation.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moulberry <james.jenour@protonmail.com>
+Date: Tue, 6 Jun 2023 02:08:29 +0800
+Subject: [PATCH] FOV Scale Implementation
+
+Modifies the player's fov by a factor
+The multiplier is affected by the client Accessibility option 'FOV Effects'
+
+diff --git a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
+index a20bcf5877106ae00c83cf376a0dd16016cd1c7c..f262347fa9de6f3f9d118a1ffa810f7292558509 100644
+--- a/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
++++ b/src/main/java/net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
+@@ -14,7 +14,7 @@ public class ClientboundPlayerAbilitiesPacket implements Packet<ClientGamePacket
+     private final boolean canFly;
+     private final boolean instabuild;
+     private final float flyingSpeed;
+-    private final float walkingSpeed;
++    public float walkingSpeed; // Paper - expose for ServerPlayer#onUpdateAbilities
+ 
+     public ClientboundPlayerAbilitiesPacket(Abilities abilities) {
+         this.invulnerable = abilities.invulnerable;
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 98df2463bf41fc736aa6a2b6ddf89e5abde6eb39..d9397ebbc2e34bed77b87d7570f8924230e1965a 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -1877,7 +1877,11 @@ public class ServerPlayer extends Player {
+     @Override
+     public void onUpdateAbilities() {
+         if (this.connection != null) {
+-            this.connection.send(new ClientboundPlayerAbilitiesPacket(this.getAbilities()));
++            // Paper start - use fov scale
++            ClientboundPlayerAbilitiesPacket abilitiesPacket = new ClientboundPlayerAbilitiesPacket(this.getAbilities());
++            abilitiesPacket.walkingSpeed = abilitiesPacket.walkingSpeed / (this.getBukkitEntity().getFovScale() * 2 - 1);
```

Comment:
The client calculates the FOV scale as:
`(movementSpeed / walkingSpeed + 1.0f) / 2.0f`

The formula here is an algebraic rearrangement to calculate the FOV scale in terms of the walkingSpeed.
I'll add an explanation of this into the patch description.

Correct me if I'm wrong, but division by 0 should be OK because this is a float, so the result is well defined to be Infinity.
Putting Infinity into the client's FOV calculation produces 0.5 as expected.